### ISSUE TITLE
StoredValue not available at launch

### ISSUE THREAD

**Issue by batjoc** [mergesort/Boutique#30]
Hi @mergesort, following up on this discussion https://github.com/mergesort/Boutique/discussions/23, I implemented the `@StoredValue` in the Performance Profiler app. However when I try to access that value on launch, it still returns `nil` and only later after the launch has been completed, it will contain the stored `RichNote`. Maybe I'm overlooking something, but I think I implemented it in the right way. You can see the code here: https://github.com/batjoc/Boutique/tree/storedvalue-nil-launch

For my own app, what I'm trying to achieve is the following: when a user opens the app and has already added an object it should show that object. Otherwise it should show the addObject screen. 

So basically in the `init` of the `ContentView`, I'm doing something like:
```
    init(model: TruckModel) {
        self.model = model
        if let truck = model.storedTruck {
            _selection = State(initialValue: .truck(truck))
        } else {
            _selection = State(initialValue: .addTruck)
        }
    }
```

But since the `model.storedTruck` always return nil, it will always show the AddObject screen instead of showing the last object the user added.

**Comment by mergesort** [comment]
@batjoc Thanks for the project and the context for how you're trying to use this! I hope this isn't blocking you too much and I'll definitely look at this but just wanted to let you know it'll be a few days since I'm on vacation right now. 🙂

**Comment by mergesort** [comment]
Sorry to put the burden on you but I'm not at a computer right now. I did take a quick look at the diff when I had a free moment and realized that the Performance Profiler project is pointing to the 2.0 release (as I mentioned in the readme), but that's incorrect advice. I've actually made quite a few changes, including enforcing that a `StoredValue` always sets a default value. Any chance you'd be able to point your project to the `main` branch and see if the issue still persists?

**Comment by batjoc** [comment]
Thanks for you quick feedback. I gave it a try with the `main` branch and although indeed in that branch there's a default value on launch, it doesn't really help me. The main problem I'm having is that I don't know what the default object should be at launch (because that's the one I'm trying to load from the `StoredValue`. So I can create a mock object, but then the app will open with a mock object that at a later stage will point at the right object, but in my case that's already too late.

For now, instead of using the `StoredValue` I simply write one object to `UserDefaults`, since it conforms to `codable`. Then this object will be available on launch while the rest of the objects load in the background.

**Comment by mergesort** [comment]
This was all really helpful and I took a step back to think about what the problem being solved here is. Given `Store` is async there's no way to solve the problem you're looking to solve, but I'm realizing that my solution isn't necessarily a solution that people are asking for. What you ended up doing, saving a value into `UserDefaults`, is what most people want, a persisted value that updates across the app when you update it from one place, available at any time even (or especially) on app launch.

To account for this what I'm going to do is create two types, `@StoredValue` and `@AsyncStoredValue`.

`@AsyncStoredValue` is what `@StoredValue` is currently (with some minor modifications). It would allow you to provide a `StorageEngine` and have a single value be persisted, with your app state still updating in real time whenever changes occur. This is useful if you want to bring your own `StorageEngine`, good for a use case like building a `KeychainStorageEngine` as I plan to do for my own apps.

The new `@StoredValue` would be very simple, and aimed to solve the problem you're trying to solve since that's what it seems people are looking to do. It would have a very similar solution to what you came up with, reading/writing data from/to `UserDefaults`. There wouldn't be a `StorageEngine` option, instead using `UserDefaults` under the hood, and the same API as `@AsyncStoredValue`.

@batjoc I assume this would solve the problem you're looking to solve, how does that solution sound to you?

**Comment by mergesort** [comment]
@batjoc The [latest release](https://github.com/mergesort/Boutique/releases/tag/2.0.0-rc2) should address this issue when you use `@StoredValue(key: "")`, with the original implementation of `@StoredValue` becoming a new type, `@AsyncStoredValue`. Can you try it out and let me know if it worked?

**Comment by batjoc** [comment]
@mergesort thanks so much for this, just tested it and it works great. This way the app can open with the last object select and if you navigate back to all objects, it shows the `items` provided by the `Store`. This is really great thanks!!

**Comment by mergesort** [comment]
I appreciate all the feedback and context you've provided me to help figure this out, and I'm glad to hear it's working for you!
